### ISSUE TITLE
How does batch normalization statistics in EMA updated when using DDP?

### ISSUE THREAD

**Issue by cyh767** [Megvii-BaseDetection/YOLOX#1176]
Hi,

I am trying to modify YOLOX code to train on my own data. I meet a problem when reading the code.

I note that EMA model is updated in this line:

https://github.com/Megvii-BaseDetection/YOLOX/blob/main/yolox/core/trainer.py#L114

However, the batch normalization statistics are synchronized after each epoch, as shown here:

https://github.com/Megvii-BaseDetection/YOLOX/blob/main/yolox/core/trainer.py#L220

So how the batch normalize statistics in EMA model synchronized across different devices in DDP mode? I think each process would update EMA model individually, but I do not find any code synchronizing the statistics coming from different processes. Please could you provide a little explanations? Thank you very much.

**Comment by FateScript** [comment]
BN on every single device is different due to [this line](https://github.com/Megvii-BaseDetection/YOLOX/blob/5bbfc110d73e80df4406e96b3c1103bb3555770f/yolox/core/trainer.py#L166) of code.
All reduce norm is used to sync batchnorm value during evaluation.
BN value of EMA model on different devices is different.

**Comment by cyh767** [comment]
I see. Thanks for your reply.
